pkg/source/ttnv2/source.go

```diff
@@ -169,6 +169,24 @@ func (s *Source) ExportDevice(devID string) (*ttnpb.EndDevice, error) {
 		}
 	}
 
+	// For OTAA devices with a session, set current parameters instead of MAC settings.
+	if deviceHasSession {
+		var err error
+		if v3dev.MACState, err = mac.NewState(v3dev, s.config.fpStore, ttnpb.MACSettings{}); err != nil {
+			return nil, err
+		}
+		// Ensure MAC state matches v2 configuration.
+		v3dev.MACState.DeviceClass = ttnpb.CLASS_A
+		v3dev.MACState.LoRaWANVersion = ttnpb.MAC_V1_0_2
+		v3dev.MACState.CurrentParameters.Rx1Delay = ttnpb.RX_DELAY_1
+		v3dev.MACState.DesiredParameters = ttnpb.MACParameters{}
```

Comment:
You should not need to be doing this. Instead, this should be covered by the field mask.
Only relevant fields in `mac_state.current_parameters` and in top-level `mac_state` should be specified